run_unittests.py

```diff
@@ -57,6 +59,39 @@ def test_version_number(self):
         self.assertEqual(searchfunc('foobar 2016.10.128'), 'unknown version')
         self.assertEqual(searchfunc('2016.10.128'), 'unknown version')
 
+    def test_mode_symbolic_to_bits(self):
+        modefunc = mesonbuild.interpreter.mode_symbolic_to_bits
+        self.assertEqual(modefunc('---------'), 0)
+        self.assertEqual(modefunc('r--------'), stat.S_IRUSR)
+        self.assertEqual(modefunc('---r-----'), stat.S_IRGRP)
+        self.assertEqual(modefunc('------r--'), stat.S_IROTH)
+        self.assertEqual(modefunc('-w-------'), stat.S_IWUSR)
+        self.assertEqual(modefunc('----w----'), stat.S_IWGRP)
+        self.assertEqual(modefunc('-------w-'), stat.S_IWOTH)
+        self.assertEqual(modefunc('--x------'), stat.S_IXUSR)
+        self.assertEqual(modefunc('-----x---'), stat.S_IXGRP)
+        self.assertEqual(modefunc('--------x'), stat.S_IXOTH)
+        self.assertEqual(modefunc('--S------'), stat.S_ISUID)
+        self.assertEqual(modefunc('-----S---'), stat.S_ISGID)
+        self.assertEqual(modefunc('--------T'), stat.S_ISVTX)
+        self.assertEqual(modefunc('--s------'), stat.S_ISUID | stat.S_IXUSR)
+        self.assertEqual(modefunc('-----s---'), stat.S_ISGID | stat.S_IXGRP)
+        self.assertEqual(modefunc('--------t'), stat.S_ISVTX | stat.S_IXOTH)
+        self.assertEqual(modefunc('rwx------'), stat.S_IRWXU)
+        self.assertEqual(modefunc('---rwx---'), stat.S_IRWXG)
+        self.assertEqual(modefunc('------rwx'), stat.S_IRWXO)
+        # We could keep listing combinations exhaustively but that seems
+        # tedious and pointless. Just test a few more.
+        self.assertEqual(modefunc('rwxr-xr-x'), stat.S_IRWXU |
+                                                stat.S_IRGRP | stat.S_IXGRP |
```

Comment:
__[flake8]__
* [E127] continuation line over-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->